src/poly/poly.jl

```diff
@@ -1131,6 +1131,15 @@ function (R::Singular.PolyRing{T})(n::T) where T<:Singular.n_unknown
    return z
 end
 
+function (R::PolyRing)(f::T) where T <: Nemo.MPolyElem
```

Comment:
This might not be relevant, but would it be a good idea to check that f doesn't already live in R?
julia> S, (a, b, c) = PolynomialRing(QQ, ["a", "b", "c"]); isa(a, Singular.Nemo.MPolyElem)
true

Reply:
Do you mean inserting something like `parent(f) == R && return f` ? That seems reasonable, as long as the polynomial is immutable, I guess?

Reply:
I checked and spolys are not immutable. Where is the problem?

Reply:
Everything is fine if you are ok with creating a copy of f when f already lives in R.

Reply:
All our other packages do not do that. If f is in R, then f is returned identically.

Reply:
I add it then as suggested by @fingolfin

Reply:
So R(f) returns f even if f is mutable? That is certainly efficient but also invites nasty to debug programming errors. Ah well. Can't have both, I guess. If that's what is done elsewhere in Oscar, fine

Reply:
It's actually done throughout Julia, though naturally it is because Julia kinda expects things to be immutable most of the time. Anyhow, what it means is that R(f) can't be used to create a copy of f. We instead use deepcopy(f). So far it doesn't seem to have caused any issues that I'm aware of.

Reply:
I had a fun time tracking down a problem caused by the return of a function f in hecke. So I did a=f(..) then later b=f(...) and found out to my horror that a and b where the same object so the second call was overwriting a. The solution was a=deepcopy(f(...)).

Reply:
I agree. I just realized, that in this case a.ptr and b.ptr are equal and this indeed can cause some trouble. 
I will add the deepcopy.

Reply:
Yes, as @wbhart said, if you need a copy, use `deepcopy`. There are some functions which are supposed to not alias the input see https://github.com/Nemocas/Nemo.jl/issues/278, but `R(a)` is not one of them.

Reply:
Then I will not add the deepcopy, if this is the convention